### PR TITLE
[ACHIEVEMENT] The label maker achievemt

### PR DESCRIPTION
**Pull request by dunaevsky**
Achievement for putting many labels in a pull request.

### CODE REVIEW

**Comment by Thatkookooguy** [comment]
@ortichon will review this so you won't get bored of my style 😉

File: achievements/labelBabyJunior.achievement.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
+
+
```
Comment (ortichon) [inline achievements/labelBabyJunior.achievement.js]: please remove the new lines above.

**Comment by dunaevsky** [comment]
Even better

File: achievements/labelBabyJunior.achievement.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+var labelBabyJunior = {
+	name: 'Label Baby Junior',
+	check: function(pullRequest, shall) {
+
```
Comment (ortichon) [inline achievements/labelBabyJunior.achievement.js]: please remove this line too :-)

notice that all(all?) other achievements follow the same design rules, It helps to keep organized and maintainable code.
I hope that someday soon we'll have a Lint process to catch and warn on this issues.

File: achievements/labelBabyJunior.achievement.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+				description: 'You\'ve put many labels, thank you for organizng. You\'re a gift that keeps on re-giving' ,
+				relatedPullRequest: pullRequest._id
+			};
+
+			shall.grant(pullRequest.creator.username, achievement);
+		}
+	}
+};
+
+function checkIfManyLabels(pullRequest) {
```
Comment (ortichon) [inline achievements/labelBabyJunior.achievement.js]: Its is a common practice to prefix with `is<FunctionName>` functions that return `boolean` values.

I think that 
```javascript
if (isManyLabels(pullRequest)) { ...
```
looks and sounds better ;-)

File: achievements/labelBabyJunior.achievement.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+	var labels = pullRequest.labels;
+	if(labels){
+		return labels.length > 3;
+	}
+	return false;
+	}
```
Comment (ortichon) [inline achievements/labelBabyJunior.achievement.js]: wrong indentation of `}`

Comment (ortichon) [inline achievements/labelBabyJunior.achievement.js]: BTW - rows 24-27 can be shorten to a one liner:
```javascript
return labels && labels.length > 3;
```

File: achievements/labelBabyJunior.achievement.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+module.exports = labelBabyJunior;
```
Comment (ortichon) [inline achievements/labelBabyJunior.achievement.js]: please add one empty line at the end of the file.
see [this answer](http://stackoverflow.com/a/2287990/4508785)

**Comment by dunaevsky** [comment]
I did. Changes considered and updated.

File: achievements/labelBabyJunior.achievement.js (hunk 4 above)
Comment (dunaevsky) [inline achievements/labelBabyJunior.achievement.js]: yea

**Comment by ortichon** [comment]
@dunaevsky Congratulations!!!!! avatar!!!!!

**Comment by dunaevsky** [comment]
Was lookin for something inline with your style ;)

File: achievements/labelBabyJunior.achievement.js [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+
+			shall.grant(pullRequest.creator.username, achievement);
+		}
+	}
+};
+
+function isManyLabels(pullRequest) {
+	var labels = pullRequest.labels;
+	return labels && labels.length > 3;
+
```
Comment (ortichon) [inline achievements/labelBabyJunior.achievement.js]: missing a `}`

please check that your code is running before you `push` to Github :-)

File: achievements/labelBabyJunior.achievement.js [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+			};
+
+			shall.grant(pullRequest.creator.username, achievement);
+		}
+	}
+};
+
+function isManyLabels(pullRequest) {
+	var labels = pullRequest.labels;
+	return labels && labels.length > 3;
```
Comment (ortichon) [inline achievements/labelBabyJunior.achievement.js]: I think that 4 labels as a minimum is a bit low.
In Kibibit-Code-editor we have at least 4 PRs with 4 labels each. 
maybe we should use `labels.length > 5` ?
and Kibibit is a small organization with a small team.

Take a look at [this PR](https://github.com/angular/angular/pull/14039) from AngularJS (Google's Javascript framework) - it has 8(!) labels. They have tons of PRs with 4-5 labels.
LMKWYT

**Comment by ortichon** [comment]
@dunaevsky 
please add `[ACHIEVEMENT] The label maker achievemt` to your pull request title.

File: achievements/labelBabyJunior.achievement.js (hunk 6 above)
Comment (dunaevsky) [inline achievements/labelBabyJunior.achievement.js]: It worked until you made me change it =P

File: achievements/labelBabyJunior.achievement.js (hunk 7 above)
Comment (dunaevsky) [inline achievements/labelBabyJunior.achievement.js]: I agree. 3 is actually just the go to num I use to test multiple stuff.

File: achievements/labelBabyJunior.achievement.js (hunk 6 above)
Comment (ortichon) [inline achievements/labelBabyJunior.achievement.js]: I did noto **made** you change it, I suggested... :-)

File: achievements/labelBabyJunior.achievement.js [hunk 8]
```diff
@@ -0,0 +1,22 @@
+var labelBabyJunior = {
+	name: 'Label Baby Junior',
+	check: function(pullRequest, shall) {
+		if (isManyLabels(pullRequest))  {
+			var achievement = {
+				avatar : 'images/achievements/theLabelMaker.achievment.jpg',
```
Comment (ortichon) [inline achievements/labelBabyJunior.achievement.js]: typo in the word `achievement` - images does not load

File: achievements/labelBabyJunior.achievement.js [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+
+			shall.grant(pullRequest.creator.username, achievement);
+		}
+	}
+};
+
+function isManyLabels(pullRequest) {
+	var labels = pullRequest.labels;
+	return labels && labels.length > 5;
+}
```
Comment (ortichon) [inline achievements/labelBabyJunior.achievement.js]: please add a new line separation between the end of the function to the `module.exports...`

File: achievements/labelBabyJunior.achievement.js [hunk 10]
```diff
@@ -0,0 +1,22 @@
+var labelBabyJunior = {
+	name: 'Label Baby Junior',
+	check: function(pullRequest, shall) {
+		if (isManyLabels(pullRequest))  {
+			var achievement = {
+				avatar : 'images/achievements/theLabelMaker.achievment.jpg',
+				name: 'The Label Maker',
+				short: 'Is this a label maker?',
+				description: 'You\'ve put many labels, thank you for organizng. You\'re a gift that keeps on re-giving' ,
```
Comment (ortichon) [inline achievements/labelBabyJunior.achievement.js]: typo in the word `organizing`

**Comment by ortichon** [comment]
Well, looks good now.
we'll wait for @Thatkookooguy's resolution about the labels bug, before approving the PR.


BTW - just a thought - don't change it, but let's discuss about it:
you've changed the image to square but did not keep the ratio, so In my eyes, Jerry looks a bit thin now, maybe we should keep the aspect ratio? (the down side is loosing some details from the sides)
what you think? 
(it's a question, I just want to hear you opinion)

**Comment by dunaevsky** [comment]
About, the aspect ratio, you're probably right. I did the easier task.

About the labels bug, I am able to reproduce it. When creating the pull request it only puts 1 label from the 6 added. But then when I try again, and open another pull request, it does add all the labels.

**Comment by Thatkookooguy** [comment]
@Kibibit/development about the bug - it worked fine for me. look at your webhook and check if github sent us all the events for the labels. Also check that they all got a `status 200` from our server. I can't see your webhook so let me know if you see something fishy

In any case, I'm going to implement this weekend another merge between the `pullRequest` object we save and the `GitHub webhook data` once a pull request is merged. That will fix some of the problem of having some data missing regardless of the issue that happen to you @ortichon

**Comment by Thatkookooguy** [comment]
I already opened a bug on github about some problems with webhooks. So if you find another one, you can report it to them

**Comment by Thatkookooguy** [comment]
I think other then fixing the aspect ratio of the image, IMO this can go in after or finishes his review

**Pull request merged**